fetch.bs

```diff
@@ -3237,7 +3250,103 @@ through TLS using ALPN. The protocol cannot be spoofed through HTTP requests in
 </ol>
 </div>
 
+<h3 id=https-upgrades>HTTPS upgrading</h3>
+
+<div algorithm>
+
+<p>User agents may optionally upgrade requests with <a>potentially untrustworthy URL</a>s
+to attempt to fetch them over <a>potentially trustworthy URL</a>s. If an upgraded request
+fails with a network error, it is retried over the original URL.
+
+<p>The HTTPS upgrading algorithm consists of Upgrade and Fallback algorithms, with the following
+steps:
+
+<h4 id=https-upgrades-upgrade>HTTPS upgrade algorithm</h4>
 
+To <dfn>upgrade an HTTP request</dfn> given a <a for=/>request</a> <var>request</var>, run these
+steps:
+
+<ol>
+  <li>
+    <p>If one or more of the following conditions are met, return:
+    <ul>
+      <li><p><var>request</var>'s <a for="request">destination</a> is not "<code>document</code>"
+
+      <li><p><var>request</var>'s <a for="request">method</a> is not "<code>GET</code>"
+
+      <li><p><var>request</var>'s <a for="request">URL</a>'s <a for="url">scheme</a> is not
+      "<code>http</code>"
+
+      <li><p><var>request</var>'s <a for="request">URL</a>'s <a for="url">host</a> is exempted from
+      upgrades in an <a>implementation-defined</a> way.
+
+      <p class="example">If <a for=url>host</a> is a non-registrable or non-assignable domain name
+      such as .local or an IP address that falls in a range reserved for non-publicly routable
+      networks, the implementation might return without modifying <var>request</var>.
+    </ul>
+  </li>
+
+  <li><p>If <var>request</var>'s <code>https-upgrade-fallback-url</code> is non-null, clear the
+  <code>https-upgrade</code> and <code>https-upgrade-fallback-url</code> flags and return.
+
+  <p class="note">This is a fallback request that shouldn't be upgraded again.
+
+  <li><p>Otherwise, upgrade the request by setting the following fields:
```

Comment:
"Upgradt the request" is not needed here. Maybe move it to a note?

Reply:
Done (I think this is okay without adding the note)